### PR TITLE
[WIP]Modify patch embed

### PR DESCRIPTION
**Pull request by HIT-cwh**
We use a conv layer to implement patch embedding. This pr allow us setting kernel_size, stride, padding and dilation used in conv layer. Pls take a look.

### CODE REVIEW

File: mmcls/models/backbones/vision_transformer.py [hunk 1]
```diff
@@ -221,41 +83,123 @@ def norm1(self):
...
+        self.embed_dims = embed_dims
+        kernal_size = to_2tuple(kernal_size)
+        stride = to_2tuple(stride)
+        padding = to_2tuple(padding)
+        dilation = to_2tuple(dilation)
+
+        patches_resolution = [
+            (self.img_size[i] + 2 * padding[i] - kernal_size[i]) // stride[i] +
+            1 for i in range(2)
+        ]
```
Comment (jshilong) [inline mmcls/models/backbones/vision_transformer.py:143]: It seems that dilation is missing (refer to [torch.nn.Unfold](https://pytorch.org/docs/stable/generated/torch.nn.Unfold.html?highlight=unfold#torch.nn.Unfold) or [torch.nn.Conv2d](https://pytorch.org/docs/stable/generated/torch.nn.Conv2d.html?highlight=conv2d#torch.nn.Conv2d)).

Comment (Junjun2016) [reply mmcls/models/backbones/vision_transformer.py:143]: It seems that dilation is missing (refer to torch.nn.Unfold or torch.nn.Conv2d).

File: mmcls/models/backbones/vision_transformer.py [hunk 2]
```diff
@@ -374,101 +321,147 @@ def forward(self, x):
...
         else:
-            self.patch_embed = PatchEmbed(
+            self.patch_embed = PatchEmbed_v2(
                 img_size=img_size,
-                patch_size=patch_size,
                 in_channels=in_channels,
-                embed_dim=embed_dim)
+                embed_dims=self.embed_dims,
+                kernal_size=patch_size,
+                stride=patch_size)
```
Comment (Junjun2016) [inline mmcls/models/backbones/vision_transformer.py:417]: We may move the args to `conv_cfg` (`patch_embed_cfg`) and it is the same in `HybridEmbed`.

File: mmcls/models/backbones/vision_transformer.py [hunk 3]
```diff
@@ -374,101 +321,147 @@ def forward(self, x):
...
+
+        self.embed_dims = self.arch_settings['embed_dims']
 
         if hybrid_backbone is not None:
             self.patch_embed = HybridEmbed(
                 hybrid_backbone,
                 img_size=img_size,
                 in_channels=in_channels,
-                embed_dim=embed_dim)
+                embed_dims=self.embed_dims)
```
Comment (Junjun2016) [inline mmcls/models/backbones/vision_transformer.py:410]: Support overlap embedding as PatchEmbed does please.

File: mmcls/models/backbones/vision_transformer.py [hunk 4]
```diff
@@ -374,101 +321,147 @@ def forward(self, x):
...
-        norm_cfg (dict): Config dict for normalization layer. Default
-            layer normalization.
-        act_cfg (dict): The activation config for FFNs. Defalut GELU.
+            Default: 0.0
+        attn_drop_rate (float): The drop out rate for attention layer.
+            Default: 0.0
+        drop_path_rate (float): stochastic depth rate. Default 0.0
+        hybrid_backbone (nn.Module, optional): CNN backbone to use in-place of
+            PatchEmbed module. Default: None
+        norm_cfg (dict): Config dict for normalization layer. Default:
```
Comment (Junjun2016) [inline mmcls/models/backbones/vision_transformer.py:343]: Add `patch_embed_cfg` and do NOT forget to modify it in Training configs.
